File: backend/src/models/post.model.js
import mongoose from "mongoose";

const MediaSchema = new mongoose.Schema(
  {
    url: { type: String, required: true },
    type: { type: String, enum: ["image", "video"], required: true },
  },
  { _id: false }
);

const PostSchema = new mongoose.Schema(
  {
    user: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },

    caption: { type: String, default: "" },
    media: { type: [MediaSchema], required: true },

    likes: [{ type: mongoose.Schema.Types.ObjectId, ref: "User" }],
    saves: [{ type: mongoose.Schema.Types.ObjectId, ref: "User" }],
    commentsCount: { type: Number, default: 0 },
  },
  { timestamps: true }
);

export const Post = mongoose.model("Post", PostSchema);

File: lib/home/post_model.dart
class Post {
  final String id;
  final String user;
  final String caption;
  final List<String> media;
  int likes;
  bool liked;
  bool saved;

  Post({
    required this.id,
    required this.user,
    required this.caption,
    required this.media,
    required this.likes,
    this.liked = false,
    this.saved = false,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      user: json['user'],
      caption: json['caption'] ?? '',
      media: List<String>.from(json['media'] ?? []),
      likes: json['likes'] ?? 0,
      liked: json['liked'] ?? false,
      saved: json['saved'] ?? false,
    );
  }
}
